src/hal/user_comps/xhc-whb04b-6/hal.cc

```diff
@@ -795,6 +796,23 @@ void Hal::clearStartResumeProgramStates()
     *memory->out.doRunProgram      = false;
     *memory->out.doResumeProgram   = false;
 }
+
+void Hal::checkState(bool state, hal_bit_t *pin) {
+	// 500 milliseconds timeout
+	unsigned int timeouts=500;
+	unsigned int timeoutMs=1;
+	do
+	{
+		if (state == *pin)
+		{
+			usleep(timeoutMs * 1000);
+		}
+		else
+		{
+			break;
+		}
+	} while ((state == *pin) && (--timeouts) > 0);
+}
```

Comment:
I don't understand this function.  It waits for the pin to change state or for a timeout to happen, but doesn't report which happened.  How is this useful?  Also, please use 4 space characters (` `) for indentation, not tab: http://linuxcnc.org/docs/2.8/html/code/style-guide.html#_indentation

Reply:
it just waits until the feedback pin changes to the requested state.
Pin A is toggled
Pin B reports the state from linuxcnc

once Pin B enters the requested state the function will stop (or time out), it could report that it didn't get the requested state.
The driver overall has several issues, I'd just commit it as it is for now and wait for further work on it to happen. There's a second patch waiting fixing other issues in the driver once this one is committed.
I'm using the handwheel on a daily basis, and I understand that there are not so many users out there (and some of them might be ok with having to push buttons several times to act without knowing why...)